test/jdk/java/foreign/TestAccessModes.java

```diff
@@ -58,9 +58,8 @@ public void testAccessModes(MemorySegment segment, MemoryLayout layout, AccessMo
         } catch (UnsupportedOperationException ex) {
             assertFalse(compatible);
         } catch (IllegalArgumentException ex) {
-            // access is unaligned, but access mode is supported
-            assertTrue(compatible ||
-                    (layout instanceof GroupLayout && segment.maxByteAlignment() < layout.byteAlignment()));
+            // access is unaligned
+            assertTrue(segment.maxByteAlignment() < layout.byteAlignment());
```

Comment:
Note: this change is required because, before this PR, we used to issue UOE for a bad access mode regardless of the alignment with which we accessed the segment (well, only for toplevel var handles). Now we uniformly check alignment _before_ access mode, for both toplevel and nested var handles, so this assertion needed to be tweaked.